bindings/rust/src/bindings/mod.rs

```diff
@@ -211,6 +273,61 @@ impl Blob {
     pub fn from_hex(hex_str: &str) -> Result<Self, Error> {
         Self::from_bytes(&hex_to_bytes(hex_str)?)
     }
+
+    pub fn samples_and_proofs(
+        &self,
+        kzg_settings: &KZGSettings,
+    ) -> Result<Vec<(Sample, KZGProof)>, Error> {
+        let samples_len = kzg_settings.max_width as usize;
+        let proofs_len = kzg_settings.sample_count as usize;
```

Comment:
In `impl KzgSettings` I think it would be better to add a few functions:

```rust
pub fn sample_count(&self) -> usize {
    self.sample_count as usize
}

pub fn sample_size(&self) -> usize {
    self.sample_size as usize
}

pub fn data_count(&self) -> usize {
    self.max_width as usize
}
```

Then use those functions.